Source/WTF/wtf/SequesteredAllocator.h

```diff
@@ -0,0 +1,579 @@
+        {
+            auto retval = alignedAllocateImpl<AllocationFailureMode::ReturnNull>(alignment, bytes);
+            dataLogFIf(verbose && (retval != nullptr),
+                "Allocator %d: Arena %d: align-allocated %luB: alloc (%p), allocHead (%p)\n",
+                m_parentSlot->id(),
+                arenaIndex(),
+                bytes,
+                retval,
+                reinterpret_cast<void*>(m_allocHead));
+            return retval;
+        }
+
+        void deallocate(void* /*p*/) { }
+
+    private:
+        static constexpr size_t minHeadAlignment = alignof(std::max_align_t);
+
+        uintptr_t headIncrementedBy(size_t bytes) const
+        {
+            const size_t alignmentMask = minHeadAlignment - 1;
+            return (m_allocHead + bytes + alignmentMask) & ~alignmentMask;
+        }
+        uintptr_t headAlignedUpTo(size_t alignment) const
+        {
+            ASSERT(alignment);
+            const size_t minHeadAlignmentMask = minHeadAlignment - 1;
+            const size_t alignmentMask = (alignment - 1) | minHeadAlignmentMask;
+            return (m_allocHead + alignmentMask) & ~alignmentMask;
+        }
+
+        template<AllocationFailureMode mode>
+        void* allocateImpl(size_t bytes)
+        {
+            uintptr_t allocation = m_allocHead;
+            uintptr_t newHead = headIncrementedBy(bytes);
+            if (newHead < m_allocBound) {
+                m_allocHead = newHead;
+                return reinterpret_cast<void*>(allocation);
+            }
+            if constexpr (!canExpand) {
+                if constexpr (mode == AllocationFailureMode::ReturnNull)
+                    return nullptr;
+                ASSERT(mode == AllocationFailureMode::Assert);
+                RELEASE_ASSERT_NOT_REACHED();
+            }
+            return allocateImplSlowPath<mode>(bytes);
+        }
+
+        template <AllocationFailureMode mode>
+        void* alignedAllocateImpl(size_t alignment, size_t bytes)
+        {
+            uintptr_t allocation = headAlignedUpTo(alignment);
+            uintptr_t newHead = headIncrementedBy((allocation - m_allocHead) + bytes);
+            if (newHead < m_allocBound) {
+                m_allocHead = newHead;
+                return reinterpret_cast<void*>(allocation);
+            }
+            if constexpr (!canExpand) {
+                if constexpr (mode == AllocationFailureMode::ReturnNull)
+                    return nullptr;
+                ASSERT(mode == AllocationFailureMode::Assert);
+                RELEASE_ASSERT_NOT_REACHED();
+            }
+            return alignedAllocateImplSlowPath<mode>(alignment, bytes);
+        }
+
+        template<AllocationFailureMode mode>
+        NEVER_INLINE void* allocateImplSlowPath(size_t bytes)
+        {
+            void* memory = mapPages<mode>(minArenaGranuleSize);
+            if constexpr (mode == AllocationFailureMode::ReturnNull) {
+                if (memory == nullptr)
```

Comment:
Let's use `!memory` for nullptr check in general. Ditto for the rest of the code.